subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientConfigurer.java

```diff
@@ -72,30 +73,47 @@
 import org.slf4j.LoggerFactory;
 
 import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
 import java.io.IOException;
 import java.net.ProxySelector;
+import java.security.NoSuchAlgorithmException;
 import java.util.Collection;
 import java.util.Collections;
 
 public class HttpClientConfigurer {
     private static final Logger LOGGER = LoggerFactory.getLogger(HttpClientConfigurer.class);
     private static final int MAX_HTTP_CONNECTIONS = 20;
-    private static final String[] SSL_PROTOCOLS;
-
+    private static final String[] SSL_PROTOCOLS = configureSslProtocols();
+    private static final String[] CIPHER_SUITES = configureCipherSuites();
     private static final String HTTPS_PROTOCOLS = "https.protocols";
+    private static final String HTTPS_CIPHER_SUITES = "https.cipherSuites";
+    private final HttpSettings httpSettings;
 
-    static {
+    private static String[] configureSslProtocols() {
         String httpsProtocols = System.getProperty(HTTPS_PROTOCOLS);
         if (httpsProtocols != null) {
-            SSL_PROTOCOLS = httpsProtocols.split(",");
+            return httpsProtocols.split(",");
         } else if (JavaVersion.current().isJava7()) {
-            SSL_PROTOCOLS = new String[]{"TLSv1", "TLSv1.1", "TLSv1.2"};
+            return new String[]{"TLSv1", "TLSv1.1", "TLSv1.2"};
         } else {
-            SSL_PROTOCOLS = null;
+            return null;
         }
     }
 
-    private final HttpSettings httpSettings;
+    private static String[] configureCipherSuites() {
+        String cipherSuites = System.getProperty(HTTPS_CIPHER_SUITES);
+        if (cipherSuites != null) {
+            return cipherSuites.split(",");
+        } else if (JavaVersion.current().isJava7()) {
+            try {
+                return SSLContext.getDefault().getSocketFactory().getSupportedCipherSuites();
```

Comment:
❌ It feels very wrong to activate weak cipher suites for everyone when downloading artifacts. The whole point of TLS 1.2 is to make things more secure. Using a weak cypher over TLS 1.2 makes no sense IMO. JCenter needs to be fixed not our code.